r1cs/gadgets/std/src/bits/uint32.rs

```diff
@@ -115,6 +116,54 @@ impl UInt32 {
         Self { value, bits }
     }
 
+    pub fn into_bits_be(self) -> Vec<Boolean> {
+        let mut ret = self.bits;
+        ret.reverse();
+        ret
+    }
+
+    pub fn from_bits_be(bits: &[Boolean]) -> Self {
+        assert_eq!(bits.len(), 32);
+
+        let mut value = Some(0u32);
+        for b in bits {
+            value.as_mut().map(|v| *v <<= 1);
+
+            match b.get_value() {
+                Some(true) => {
+                    value.as_mut().map(|v| *v |= 1);
+                }
+                Some(false) => {}
+                None => {
+                    value = None;
+                }
+            }
+        }
+
+        UInt32 {
+            value,
+            bits: bits.iter().rev().cloned().collect(),
+        }
+    }
+
+    pub fn rotl(&self, by: usize) -> Self {
+        let by = by % 32;
+
+        let new_bits = self
+            .bits
+            .iter()
+            .skip(32 - by)
```

Comment:
So basically here, you are rotating right by 32-by, which is equivalent. Wasn't shorter to write
return rotr(32-by)?